.github/workflows/deps.yaml

```diff
@@ -0,0 +1,100 @@
+name: Check Compability
+on:
+  - pull_request
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+env:
+  UV_SYSTEM_PYTHON: true
+jobs:
+  check-requests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        requests-version:
+          - '2.32.3'
+          - '2.32.2'
+          - '2.31.0'
+          - '2.30.0'
+          - '2.29.0'
+          - '2.28.2'
+          - '2.28.1'
+          - '2.28.0'
+          - '2.27.1'
+          - '2.27.0'
+          - '2.26.0'
+          - '2.25.1'
+          - '2.25.0'
+          - '2.24.0'
+          - '2.23.0'
+          - '2.22.0'
+          - '2.21.0'
+          - '2.20.1'
+          - '2.20.0'
+          - '2.19.1'
+          - '2.19.0'
+          - '2.18.4'
+          - '2.18.3'
+          - '2.18.2'
+          - '2.18.1'
+          - '2.18.0'
+          - '2.17.3'
+          - '2.17.2'
+          - '2.17.1'
+          - '2.17.0'
+          - '2.16.5'
+          - '2.16.4'
+          - '2.16.3'
+          - '2.16.2'
+          - '2.16.1'
+          - '2.16.0'
+          - '2.15.1'
+          - '2.15.0'
+          - '2.14.2'
+          - '2.14.1'
+          - '2.14.0'
+          - '2.13.0'
+          - '2.12.5'
+          - '2.12.4'
+          - '2.12.3'
+          - '2.12.2'
+          - '2.12.1'
+          - '2.12.0'
+          - '2.11.1'
+          - '2.11.0'
+          - '2.10.0'
+          - '2.9.2'
+          - '2.9.1'
+          - '2.9.0'
+          - '2.8.1'
+          - '2.8.0'
+          - '2.7.0'
+          - '2.6.2'
+          - '2.6.1'
+          - '2.6.0'
+          - '2.5.3'
+          - '2.5.2'
+          - '2.5.1'
+          - '2.5.0'
+          - '2.4.3'
+          - '2.4.2'
+          - '2.4.1'
+          - '2.4.0'
+          - '2.3.0'
+          - '2.2.1'
+          - '2.2.0'
+          - '2.1.0'
+          - '2.0.1'
+          - '2.0.0'
```

Comment:
This feels like overkill. Why not just test on 2.0.0, and if we find that we need to add a specific version later, we can?

Reply:
You're probably right. This is just an attempt at understanding where things break.  This latest commit is broken, but if you look [here](https://github.com/posit-dev/posit-sdk-py/actions/runs/11348910107), you will see that things break starting with 2.27.1 and earlier. This issue stems from how we are catching errors in the hooks module.